src/mmw/apps/geoprocessing_api/calcs.py

```diff
@@ -135,8 +135,33 @@ def catchment_water_quality(geojson):
             ]
         else:
             catchment_water_quality_results = []
+
+    def catchment_intersects_aoi(aoi, geom):
```

Comment:
I'd recommend taking this function out of the containing function, as it's a little hard to follow the flow of the outer function.  The var `geom` is declared in both the inner and outer function, and in the outer scope references to the aoi, but in the inner scope references an individual catchment result.  I think, given that there are multiple geometries being handled here, we should abandon `geom` as a var name since it's not clear which it refers to.

Reply:
Sounds good -- I moved this out into a separate function with better argument names, and also passed in a cloned version of the area of interest geometry in 5070.